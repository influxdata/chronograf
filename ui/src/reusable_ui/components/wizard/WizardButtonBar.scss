/*
  Wizard Button Bar Styles
  ------------------------------------------------------------------------------
*/

.wizard-button-bar {
  display: flex;
  margin: $ix-marg-b auto $ix-marg-c auto;
  position: relative;
  min-width: 100%;
  justify-content: center;
  align-items: center;
  height: 50px;
  .btn {
    position: relative;
    white-space: nowrap;
    margin: 0 $ix-marg-a;
  }
}
