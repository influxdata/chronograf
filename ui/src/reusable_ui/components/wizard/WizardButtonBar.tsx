// Libraries
import React, {PureComponent} from 'react'

import {ErrorHandling} from 'src/shared/decorators/errors'

interface Props {
  decrement?: () => void
  nextLabel?: string
  previousLabel?: string
  lastStep?: boolean
  onClickPrevious: () => void
  onClickNext: () => void
}

@ErrorHandling
class WizardButtonBar extends PureComponent<Props> {
  public static defaultProps: Partial<Props> = {
    nextLabel: 'next',
    previousLabel: 'previous',
  }

  public render() {
    const {
      decrement,
      previousLabel,
      nextLabel,
      onClickPrevious,
      onClickNext,
    } = this.props
    return (
      <div className="wizard-button-bar">
        {decrement && (
          <button className="btn btn-md btn-default" onClick={onClickPrevious}>
            {previousLabel}
          </button>
        )}
        <button
          className={`btn btn-md ${this.buttonColor}`}
          onClick={onClickNext}
        >
          {nextLabel}
        </button>
      </div>
    )
  }

  private get buttonColor() {
    const {lastStep} = this.props

    if (lastStep) {
      return 'btn-success'
    }

    return 'btn-primary'
  }
}

export default WizardButtonBar
