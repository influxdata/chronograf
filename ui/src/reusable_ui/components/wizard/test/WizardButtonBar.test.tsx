import React from 'react'
import {shallow} from 'enzyme'

import WizardButtonBar from 'src/reusable_ui/components/wizard/WizardButtonBar'

describe('WizardButtonBar', () => {
  let wrapper

  const props = {
    onClickPrevious: jest.fn(),
    onClickNext: jest.fn(),
    decrement: undefined,
    nextLabel: undefined,
    previousLabel: undefined,
    lastStep: undefined,
  }

  beforeEach(() => (wrapper = shallow(<WizardButtonBar {...props} />)))

  it('mounts without exploding', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('renders a div', () => {
    expect(wrapper.find('div')).toHaveLength(1)
  })

  it('renders only the "Next" button if decrement is undefined', () => {
    const button = wrapper.find('button')
    expect(button).toHaveLength(1)
    expect(button.props().children).toBe('Next')
  })

  it('renders a primary button if lastStep is falsy', () => {
    const buttonColor = wrapper.instance().buttonColor
    expect(buttonColor).toBe('btn-primary')
  })

  it('matches snapshot when provided minimal props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('with decrement and last step is "true"', () => {
    const propsWithDecrement = {
      ...props,
      decrement: jest.fn(),
      lastStep: true,
    }

    beforeEach(() => {
      jest.resetAllMocks()
      wrapper = shallow(<WizardButtonBar {...propsWithDecrement} />)
    })

    it('renders the "Previous" button if decrement is defined and lastStep is true', () => {
      const button = wrapper.find('button')
      expect(button).toHaveLength(2)
      expect(button.at(0).props().children).toBe('Previous')
    })

    it('renders a success button if lastStep is true', () => {
      const buttonColor = wrapper.instance().buttonColor
      expect(buttonColor).toBe('btn-success')
    })

    it('matches snapshot when provided decrement function and last step as "true"', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
