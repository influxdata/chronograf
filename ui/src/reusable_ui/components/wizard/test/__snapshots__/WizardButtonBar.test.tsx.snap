// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardButtonBar matches snapshot when provided minimal props 1`] = `
<div
  className="wizard-button-bar"
>
  <button
    className="btn btn-md btn-primary"
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;

exports[`WizardButtonBar with decrement and last step is "true" matches snapshot when provided decrement function and last step as "true" 1`] = `
<div
  className="wizard-button-bar"
>
  <button
    className="btn btn-md btn-default"
    onClick={[MockFunction]}
  >
    Previous
  </button>
  <button
    className="btn btn-md btn-success"
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
